src/overlays/actors/ovl_En_Light/z_en_light.c

```diff
@@ -21,15 +27,181 @@ const ActorInit En_Light_InitVars = {
+    f32 intensity;
+    FlameParams* flameParams;
+    s16 radius;
+    EnLight* this = THIS;
+
+    flameParams = &D_80A9E840[this->actor.params & 0xF];
+    intensity = (Math_Rand_ZeroOne() * 0.5f) + 0.5f;
+    radius = (this->actor.params < 0) ? 100 : 300;
+    Lights_SetPositionalLightColorAndRadius(&this->posLightInfo, (flameParams->primColor.r * intensity),
+                                            (flameParams->primColor.g * intensity),
+                                            (flameParams->primColor.b * intensity), radius);
+    EnLight_UpdatePosRot(this, globalCtx);
+
+    if (this->actor.params >= 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+}
+
+void EnLight_UpdateSwitch(Actor* thisx, GlobalContext* globalCtx) {
+    f32 intensity;
+    FlameParams* flameParams;
+    EnLight* this = THIS;
+    f32 scale;
+
+    flameParams = &D_80A9E840[this->actor.params & 0xF];
+    scale = this->actor.scale.x / ((f32)flameParams->scale * 0.0001);
+
+    if (this->actor.params & 0x800) {
+        if (Flags_GetSwitch(globalCtx, (this->actor.params & 0x3F0) >> 4)) {
+            Math_ApproxF(&scale, 1.0f, 0.05f);
+        } else {
+            if (scale < 0.1f) {
+                Actor_SetScale(&this->actor, 0.0f);
+                return;
+            }
+            Math_ApproxF(&scale, 0.0f, 0.05f);
+        }
+    } else {
+        if (Flags_GetSwitch(globalCtx, (this->actor.params & 0x3F0) >> 4)) {
+            if (scale < 0.1f) {
+                Actor_SetScale(&this->actor, 0.0f);
+                return;
+            }
+            Math_ApproxF(&scale, 0.0f, 0.05f);
+        } else {
+            Math_ApproxF(&scale, 1.0f, 0.05f);
+        }
+    }
+
+    Actor_SetScale(&this->actor, ((f32)flameParams->scale * 0.0001) * scale);
+    intensity = (Math_Rand_ZeroOne() * 0.5f) + 0.5f;
+    Lights_SetPositionalLightColorAndRadius(&this->posLightInfo, (flameParams->primColor.r * intensity),
+                                            (flameParams->primColor.g * intensity),
+                                            (flameParams->primColor.b * intensity), 300.0f * scale);
+    EnLight_UpdatePosRot(this, globalCtx);
+
+    if (this->actor.params >= 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+}
+
+void EnLight_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+    s32 pad;
+    FlameParams* flameParams;
+    Gfx* dList;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
+
+    if (1) {}
+
+    flameParams = &D_80A9E840[this->actor.params & 0xF];
+
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(&dispRefs, globalCtx->state.gfxCtx, "../z_en_light.c", 441);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    if (this->actor.params >= 0) {
+        gSPSegment(
+            gfxCtx->polyXlu.p++, 0x08,
+            Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 32, 64, 1, 0, (this->timer * -20) & 0x1FF, 32, 128));
```

Comment:
```suggestion
            Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 32, 64, 1, 0, (this->timer * -20) % 512, 32, 128));
```

Reply:
mod dont work, you get the compiler generated nonsense

Reply:
Does `% 512U` work?

Reply:
it does